hknweb/exams/models.py

```diff
@@ -1,30 +1,39 @@
 from django.db import models
 
 class Department(models.Model):
-    name = models.CharField(unique=True, max_length=4, null=True) #short form (e.g. 'EE')
-    subject = models.CharField(max_length=255) #long form (e.g. 'Electrical Engineering')
+    abbreviated_name = models.CharField(unique=True, max_length=4, null=True) #short form (e.g. 'EE')
+    long_name = models.CharField(max_length=255) #long form (e.g. 'Electrical Engineering')
+
+    def getAbbrName(self):
+        return self.abbreviated_name
 
     def __str__(self):
-        return "{}".format(self.subject)
+        return "{} ({})".format(self.long_name, self.abbreviated_name)
 
 class Instructor(models.Model):
     name = models.CharField(max_length=255)
     department = models.ForeignKey(Department, on_delete=models.CASCADE, null=False)
 
     def __str__(self):
-        return "Instructor(name={}, department={})".format(self.name, self.department)
+        return "{}".format(self.name) #"Instructor(name={}, department={})".format(self.name, self.department)
 
 class Course(models.Model):
     name        = models.CharField(max_length=255, null=False)
     department  = models.ForeignKey(Department, on_delete=models.CASCADE, null=False)
     number 		= models.CharField(max_length=10, null=False)
 
     def __str__(self):
-        return "{} {}".format(self.department, self.number)
+        return "{} {}".format(self.department.getAbbrName(), self.number)
+
+class Semester(models.Model):
+     semester = models.CharField(max_length=255)
+
+     def __str__(self):
+         return "{}".format(self.semester)
 
 class CourseSemester(models.Model):
```

Comment:
Nitpick, but why did we call this a `CourseSemester` if its purpose is to hold the exams?

Reply:
Means: Courses, but linked to Semesters (and Exams as well)

Compared to Courses, which is just a class with no Semesters linked to them